pkg/inventory/web/client.go

```diff
@@ -35,11 +35,46 @@ type EventHandler interface {
 	// Resource deleted.
 	Deleted(r Event)
 	// An error has occurred.
-	Error(error)
+	// The handler may call the Repair() on
+	// the watch to repair the watch as desired.
+	Error(*Watch, error)
```

Comment:
Pass the error **and the watch** to the handler.
This delegates responsibility of remediation to the handler. The handler can call Watch.Repair() or Watch.End() as desired.